### PR TITLE
Add new API docs for System.Runtime.InteropServices

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Adds the rest of the missing API docs we can put in code from #105986

@carlossanlop the CollectionsMarshal APIs do have docs in source. Do we need to manually port them to dotnet-api-docs?

### CODE REVIEW

**Comment by carlossanlop** [comment]
> the CollectionsMarshal APIs do have docs in source. Do we need to manually port them to dotnet-api-docs?

@jkoritzinsky  Yes, they need them manually ported, because the source of truth is not triple slash comments, but dotnet-api-docs, as indicated by this turned-off property in the csproj:

https://github.com/dotnet/runtime/blob/d756da3f1d641ac528d696475cc8b244f6b58b0e/src/libraries/System.Runtime.InteropServices/src/System.Runtime.InteropServices.csproj#L7

**Comment by carlossanlop** [comment]
Note: We can't just turn the property on, because we would need to backport the language-reviewed docs from dotnet-api-docs into runtime first.

File: src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComVariantMarshaller.cs [hunk 1]
```diff
@@ -22,6 +22,13 @@ public static partial class ComVariantMarshaller
         // VARIANT_BOOL constants.
         private const short VARIANT_TRUE = -1;
         private const short VARIANT_FALSE = 0;
+
+        /// <summary>
+        /// Convert a managed object to an unmanaged <see cref="ComVariant"/>.
+        /// </summary>
+        /// <param name="managed">The managed object</param>
```
Comment (carlossanlop) [inline src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComVariantMarshaller.cs:29]:
```suggestion
        /// <param name="managed">The managed object.</param>
```

**Comment by jkoritzinsky** [comment]
Okay I'll port the docs over manually after this PR gets in.

File: src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComVariantMarshaller.cs [hunk 2]
```diff
@@ -103,6 +110,12 @@ private static unsafe bool TryCreateOleVariantForInterfaceWrapper(object managed
         }
 #pragma warning restore CA1416 // Validate platform compatibility
 
+        /// <summary>
+        /// Convert an unmanaged <see cref="ComVariant"/> to a managed object.
+        /// </summary>
+        /// <param name="unmanaged">The unmanaged variant</param>
```
Comment (carlossanlop) [inline src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComVariantMarshaller.cs:116]:
```suggestion
        /// <param name="unmanaged">The unmanaged variant.</param>
```

File: src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComVariantMarshaller.cs [hunk 3]
```diff
@@ -195,6 +208,10 @@ private static unsafe bool TryCreateOleVariantForInterfaceWrapper(object managed
 #pragma warning restore CS0618 // Type or member is obsolete
         }
 
+        /// <summary>
+        /// Dispose the unmanaged <see cref="ComVariant"/>.
```
Comment (carlossanlop) [inline src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComVariantMarshaller.cs:212]:
```suggestion
        /// Disposes the unmanaged <see cref="ComVariant"/>.
```

File: src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComVariantMarshaller.cs [hunk 4]
```diff
@@ -103,6 +110,12 @@ private static unsafe bool TryCreateOleVariantForInterfaceWrapper(object managed
         }
 #pragma warning restore CA1416 // Validate platform compatibility
 
+        /// <summary>
+        /// Convert an unmanaged <see cref="ComVariant"/> to a managed object.
```
Comment (carlossanlop) [inline src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComVariantMarshaller.cs:114]:
```suggestion
        /// Converts an unmanaged <see cref="ComVariant"/> to a managed object.
```

File: src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComVariantMarshaller.cs [hunk 5]
```diff
@@ -22,6 +22,13 @@ public static partial class ComVariantMarshaller
         // VARIANT_BOOL constants.
         private const short VARIANT_TRUE = -1;
         private const short VARIANT_FALSE = 0;
+
+        /// <summary>
+        /// Convert a managed object to an unmanaged <see cref="ComVariant"/>.
```
Comment (carlossanlop) [inline src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComVariantMarshaller.cs:27]:
```suggestion
        /// Converts a managed object to an unmanaged <see cref="ComVariant"/>.
```

**Review by carlossanlop — APPROVED** [review]
Left some suggestions for you to consider, and aside from that, it looks good.

**Pull request merged**